processor/transformprocessor/README.md

```diff
@@ -37,9 +37,9 @@ the fields specified by the list of strings. e.g., `keep_keys(attributes, "http.
 
 - `limit(target, limit)` - `target` is a path expression to a map type field. `limit` is a non-negative integer.  The map will be mutated such that the number of items does not exceed the limit. e.g., `limit(attributes, 100)` will limit `attributes` to no more than 100 items. Which items are dropped is random.
 
-- `replace_match(target, pattern, replacement)` - `target` is a path expression to a telemetry field, `pattern` is a string following [filepath.Match syntax](https://pkg.go.dev/path/filepath#Match), and `replacement` is a string. If `target` matches `pattern` it will get replaced with `replacement`. e.g., `replace_match(attributes["http.target"], "/user/*/list/*", "/user/{userId}/list/{listId}")`
+- `replace_match(target, regex, replacement)` - `target` is a path expression to a telemetry field, `regex` is a regex string indicating a segment to replace, and `replacement` is a string. If sections of `target` match `regex` they will get replaced with `replacement`. e.g., `replace_match(resource.attributes["process.command_line"], "password\\=[^\\s]*(\\s?)", "password=***")`
```

Comment:
I like the idea of the regex being the exact section to match, but I didn't realize that was your intention when we were discussing in the issue.  That ends up significantly changing how the function behaves.  Before if there was a match then the supplied replacement was taken verbatim, but this version has more nuance to it.  Since there is only 1 replacement value you would no longer be able to do something like `replace_match(attributes["http.target"], "/user/*/list/*", "/user/{userId}/list/{listId}"` which is a capability we definitely need to keep.

`replace_match` and `replace_all_matches` can still be updated to use regex instead of glob, but the replacement string needs to be the full replacement.  

The new concept you've proposed for replacing all occurrences of a match with a single value is also valuable and should be a new function.  The new function would not deprecate the 2 existing functions.

Reply:
Ok, I'll leave replace_match and replace_all_matches exactly as they are now.  With regard to the new functions, how do you feel about replace_pattern and replace_all_pattern for the functions I added?

Reply:
Yes I like those names.